src/puyapy/awst_build/eb/tuple.py

```diff
@@ -133,14 +134,17 @@ def call(
         if any_missing:
             return dummy_value(pytype, location)
 
-        # TODO: need to evaluate values args in field_mapping order
-        values = [
-            expect.argument_of_type_else_dummy(field_mapping[field_name], field_type).resolve()
-            for field_name, field_type in pytype.fields.items()
-        ]
-        expr = TupleExpression(
-            items=values,
+        expr = NamedTupleExpression(
             wtype=pytype.wtype,
+            values=[
```

Comment:
This works because of the attrs field converter, but to be closer to the true type you can just use a dictionary comprehension here. 

Also I'd suggest having the dictionary comprehension as it's own statement for readability.